pkg/setting/setting.go

```diff
@@ -724,35 +727,44 @@ func pathExists(path string) bool {
 	return false
 }
 
-func setHomePath(args *CommandLineArgs) {
+func (cfg *Cfg) setHomePath(args CommandLineArgs) {
 	if args.HomePath != "" {
-		HomePath = args.HomePath
+		cfg.HomePath = args.HomePath
+		HomePath = cfg.HomePath
 		return
 	}
 
 	var err error
-	HomePath, err = filepath.Abs(".")
+	cfg.HomePath, err = filepath.Abs(".")
 	if err != nil {
 		panic(err)
 	}
+
+	HomePath = cfg.HomePath
 	// check if homepath is correct
-	if pathExists(filepath.Join(HomePath, "conf/defaults.ini")) {
+	if pathExists(filepath.Join(cfg.HomePath, "conf/defaults.ini")) {
 		return
 	}
 
 	// try down one path
-	if pathExists(filepath.Join(HomePath, "../conf/defaults.ini")) {
-		HomePath = filepath.Join(HomePath, "../")
+	if pathExists(filepath.Join(cfg.HomePath, "../conf/defaults.ini")) {
+		cfg.HomePath = filepath.Join(cfg.HomePath, "../")
+		HomePath = cfg.HomePath
 	}
 }
 
 var skipStaticRootValidation = false
 
-func NewCfg() *Cfg {
-	return &Cfg{
+func NewCfg(args CommandLineArgs) (*Cfg, error) {
```

Comment:
This change is creating a lot of problems in many tests. Possible to keep `NewCfg()` and introduce a new function that takes args and loads the config?

Reply:
@marefr that sounds fine to me.